public/scripts/extensions/regex/index.js

```diff
@@ -526,14 +541,24 @@ async function saveRegexScript(regexScript, existingScriptIndex, scriptType, sav
     }
 
     if (scriptType === SCRIPT_TYPES.SCOPED) {
-        await writeExtensionField(this_chid, 'regex_scripts', array);
+        await saveScriptsByType(array, SCRIPT_TYPES.SCOPED);
 
         // Add the character to the allowed list
         if (!extension_settings.character_allowed_regex.includes(characters[this_chid].avatar)) {
             extension_settings.character_allowed_regex.push(characters[this_chid].avatar);
         }
     }
 
+    if (scriptType === SCRIPT_TYPES.PRESET) {
+        await saveScriptsByType(array, SCRIPT_TYPES.PRESET);
+
+        // Add the preset to the allowed list
+        const presetName = getPresetName();
+        if (!extension_settings.preset_allowed_regex[main_api].includes(presetName)) {
```

Comment:
Here too, I would add the null check at least that `preset_allowed_regex` exists.

```suggestion
        if (!extension_settings.preset_allowed_regex?.[main_api].includes(presetName)) {
```